### PR TITLE
Use network strings to ask the client to run a command

### PR DESCRIPTION
**Pull request by brandonsturgeon**

### CODE REVIEW

File: lua/autorun/sh_autoclean.lua [hunk 1]
```diff
@@ -0,0 +1,15 @@
+--[[
+-- TODO: Find a way to get this type of structure to work
+-- Ideally we'd have one place that both the client and server can read the clientside commands
```
Comment (brandonsturgeon) [inline lua/autorun/sh_autoclean.lua]: We'd also store shared strings like the netmessage string

File: lua/autorun/client/cl_autoclean.lua [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+local function notifyPlayer( notification )
+    local message = "[CFC_AutoClean] " .. notification
+
+    print( message )
+end
+
+net.Receive( "CFC_AutoClean_RunCommand", function() 
+    local command = net.ReadString()
+
+    if isAllowedCommand( command ) then
```
Comment (brandonsturgeon) [inline lua/autorun/client/cl_autoclean.lua]: Extra bit of security for the client

Comment (PLally) [inline lua/autorun/client/cl_autoclean.lua]:
```suggestion
    if isAllowedCommand[ command ] then
```

Comment (brandonsturgeon) [reply lua/autorun/client/cl_autoclean.lua]: Good catch, thanks.

Comment (brandonsturgeon) [reply lua/autorun/client/cl_autoclean.lua]: Good catch, thanks.

File: lua/autorun/server/sv_autoclean.lua [hunk 3]
```diff
@@ -43,10 +43,16 @@ function cfcCleanServer()
...
+    runCleanupCommandsOnPlayers()
+    removeUnownedWeapons()
 end
 
-timer.Create("cfc_Autoclean", GetConVar("cfc_autoclean"):GetInt(), 0, cfcCleanServer)
+timer.Create( "cfc_Autoclean", GetConVar( "cfc_autoclean" ):GetInt(), 0, runCleanupFunctions )
 
-hook.Remove("cfc_CleanOnLag")
-hook.Add("APG_lagDetected", "cfc_CleanOnLag", cfcCleanServer)
+hook.Remove( "APG_lagDetected", "CFC_CleanOnLAg" )
```
Comment (PLally) [inline lua/autorun/server/sv_autoclean.lua]:
```suggestion
hook.Remove( "APG_lagDetected", "CFC_CleanOnLag" )
```

File: lua/autorun/server/sv_autoclean.lua [hunk 4]
```diff
@@ -43,10 +43,16 @@ function cfcCleanServer()
...
+    notifyPlayers( message )
+end
+
+local function runCleanupFunctions()
+    runCleanupCommandsOnPlayers()
+    removeUnownedWeapons()
 end
 
-timer.Create("cfc_Autoclean", GetConVar("cfc_autoclean"):GetInt(), 0, cfcCleanServer)
+timer.Create( "cfc_Autoclean", GetConVar( "cfc_autoclean" ):GetInt(), 0, runCleanupFunctions )
```
Comment (brandonsturgeon) [inline lua/autorun/server/sv_autoclean.lua]:
```suggestion
timer.Create( "CFC_AutoClean", GetConVar( "cfc_autoclean" ):GetInt(), 0, runCleanupFunctions )
```

**Pull request merged**